src/NHSD.BuyingCatalogue.Ordering.Api/Services/CreatePurchasingDocument/AttachmentCsvWriter.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading.Tasks;
+using CsvHelper;
+using CsvHelper.Configuration;
+using CsvHelper.TypeConversion;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.Services.CreatePurchasingDocument
+{
+    public sealed class AttachmentCsvWriter<T, T2> : IAttachmentCsvWriter<T> where T2 : ClassMap<T>
```

Comment:
Could the T's get a little more context, just to help with understanding on first review. e.g T = TEntity / TRecord & T2 = TConfig / TClassMap

Reply:
The context of this class is to wrap the CsvWriter functionality. Just thinking of a name change, what do you think to `CsvStreamWriter`? Just because the scenario we are using it in at the moment is for an attachment, I think am I right in saying it could be used to create a file for example?

Reply:
Yep, you're right 😄